From 55cc049c8aa3acb7e69d17edbb5a033a9896f8dd Mon Sep 17 00:00:00 2001
From: Daniel Steinmetzer <dsteinmetzer@seemoo.tu-darmstadt.de>
Date: Tue, 5 Sep 2017 09:48:53 +0200
Subject: [PATCH 55/59] seemoo: move sweep dump declarations to external file

Signed-off-by: Daniel Steinmetzer <dsteinmetzer@seemoo.tu-darmstadt.de>
---
 wil6210/debugfs.c | 30 ++----------------------------
 1 file changed, 2 insertions(+), 28 deletions(-)

diff --git a/drivers/net/wireless/ath/wil6210/debugfs.c b/drivers/net/wireless/ath/wil6210/debugfs.c
index 531f3ec..aab7932 100644
--- a/drivers/net/wireless/ath/wil6210/debugfs.c
+++ b/drivers/net/wireless/ath/wil6210/debugfs.c
@@ -24,6 +24,8 @@
 #include "wmi.h"
 #include "txrx.h"
 #include "pmc.h"
+/* SEEMOO shared header between driver and fw */
+#include "sweep_info.h"
 
 /* Nasty hack. Better have per device instances */
 static u32 mem_addr;
@@ -51,29 +53,6 @@ struct dbg_off {
 #define MY_SECTOR_TABLE_SIZE 16
 #define SWEEP_DUMP_SIZE 256
 
-typedef struct {
-        u32 sector_id;
-        u32 crssi;
-        s32 snr;
-} sector_info_t;
-
-typedef struct {
-	u32 ctr_valid;
-	u32 ctr_overflow;
-	u16 ctr_null;
-	u16 ctr_missed;
-	u8 reserved;
-	u8 flag;
-	u16 reserved2;
-} sweep_stats_t;
-
-typedef struct {
-        u32 cur_pos;
-	u32 ctr_pkts;
-	u32 ctr_swps;
-        sector_info_t dump[SWEEP_DUMP_SIZE];
-} sweep_dump_t;
-
 #define MY_CONSOLE_MAX_LEN 8192
 u8 my_console_fw[MY_CONSOLE_MAX_LEN];
 u8 my_console_uc[MY_CONSOLE_MAX_LEN];
@@ -81,11 +60,6 @@ u32 my_console_fw_len;
 u32 my_console_uc_len;
 #define MY_CONSOLE_BASE_PTR_FW 0x8ffff0
 #define MY_CONSOLE_BASE_PTR_UC 0x935ff4
-#define MY_SWEEP_INFO_PTR 0x93D000
-
-#define PTR_SWEEP_STATS 0x9406FC
-#define PTR_MEM_SWEEP_CUR_FEEDBACK 0x93D000
-#define PTR_MEM_SWEEP_DUMP 0x93D018
 
 struct wil6210_priv *my_glob_wil = NULL;
 
-- 
2.14.3 (Apple Git-98)

