From d29e4490178459cfd71c145a66712a5cffe07009 Mon Sep 17 00:00:00 2001
From: Daniel Steinmetzer <dsteinmetzer@seemoo.tu-darmstadt.de>
Date: Tue, 5 Sep 2017 15:15:56 +0200
Subject: [PATCH 58/59] seemoo: enable debug_fs compilation by default again

Signed-off-by: Daniel Steinmetzer <dsteinmetzer@seemoo.tu-darmstadt.de>
---
 wil6210/Makefile  |  2 +-
 wil6210/wil6210.h | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/wil6210/Makefile b/wil6210/Makefile
index 8a19dbc..76267d4 100644
--- a/wil6210/Makefile
+++ b/wil6210/Makefile
@@ -4,7 +4,7 @@ wil6210-y := main.o
 wil6210-y += netdev.o
 wil6210-y += cfg80211.o
 wil6210-y += pcie_bus.o
-wil6210-$(CONFIG_WIL6210_DEBUGFS) += debugfs.o
+wil6210-y += debugfs.o
 wil6210-y += wmi.o
 wil6210-y += interrupt.o
 wil6210-y += txrx.o
diff --git a/wil6210/wil6210.h b/wil6210/wil6210.h
index 8b17d64..01295f0 100644
--- a/wil6210/wil6210.h
+++ b/wil6210/wil6210.h
@@ -951,13 +951,13 @@ int wil_cfg80211_mgmt_tx(struct wiphy *wiphy, struct wireless_dev *wdev,
 			 struct cfg80211_mgmt_tx_params *params,
 			 u64 *cookie);
 
-#if defined(CONFIG_WIL6210_DEBUGFS)
+// #if defined(CONFIG_WIL6210_DEBUGFS)
 int wil6210_debugfs_init(struct wil6210_priv *wil);
 void wil6210_debugfs_remove(struct wil6210_priv *wil);
-#else
-static inline int wil6210_debugfs_init(struct wil6210_priv *wil) { return 0; }
-static inline void wil6210_debugfs_remove(struct wil6210_priv *wil) {}
-#endif
+// #else
+// static inline int wil6210_debugfs_init(struct wil6210_priv *wil) { return 0; }
+// static inline void wil6210_debugfs_remove(struct wil6210_priv *wil) {}
+// #endif
 
 int wil_cid_fill_sinfo(struct wil6210_priv *wil, int cid,
 		       struct station_info *sinfo);
-- 
2.14.3 (Apple Git-98)

